themes-default/slim/src/store/modules/shows.js

```diff
@@ -84,17 +85,40 @@ const actions = {
     getShows(context, shows) {
         const { commit, dispatch } = context;
 
-        // If no shows are provided get the first 10k
+        // If no shows are provided get the first 1000
         if (!shows) {
-            const params = {
-                limit: 10000
-            };
-            return api.get('/series', { params }).then(res => {
-                const shows = res.data;
-                return shows.forEach(show => {
+            return (async() => {
+                const limit = 1000;
+                const page = 1;
+                const params = {
+                    limit,
+                    page
+                };
+
+                // Get first page
+                try {
+                    const response = await api.get('/series', { params });
+                } catch {
+                    console.log(
+                        'Could not retrieve a list of shows'
+                    );
+                    return;
+                }
+                const totalPages = Number(response.headers['x-pagination-total']);
+                response.data.forEach(show => {
                     commit(ADD_SHOW, show);
                 });
-            });
+
+                // Optionally get additional pages
+                for (let x = 2; x <= totalPages; x++) {
+                    params.page = x;
+                    api.get('/series', { params }).then(response => {
```

Comment:
Disclaimer - I did make some of the changes I proposed with `@babel/polyfill` above, but nothing on to this part of the code.
I just wanted to let you know it doesn't seem to work for me.

![image](https://user-images.githubusercontent.com/10238474/51167867-5d4aba80-18b0-11e9-8b6c-6862c86bef46.png)

Reply:
Yeah i've fixed this. IT's also now using promises. So I can park the async/await discussion for now.